lib/chef/provider/cron.rb

```diff
@@ -51,6 +67,9 @@ def load_current_resource
         @current_resource = Chef::Resource::Cron.new(@new_resource.name)
         @current_resource.user(@new_resource.user)
         @cron_exists = false
+        # Raise an execption if Chef can't read or update the crontab.
+        can_modify_cron?
```

Comment:
@lamont-granquist Currently playing with the assertions, from the provider.rb the resource requirement are processed after the load_current_resource, in case of cron, the read_crontab (which this PR aim to improve) is called within load_current_resource, and the validation has to be done before. I don't see how to achieve this properly :/ I think I see how to add why-run support to the existing code, but I don't see how moving to d_r_r could help. Maybe using the code above in `check_resource_semantics!` and raising or messaging if in why-run ?